src/main/java/controller/user/DeleteUserServlet.java

```diff
@@ -0,0 +1,45 @@
+package controller.user;
+
+import dao.impl.UserDaoImpl;
+import factory.UserServiceFactory;
+import model.User;
+import org.apache.log4j.Logger;
+import service.UserService;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+@WebServlet(value = "/delete/user")
+public class DeleteUserServlet extends HttpServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(UserDaoImpl.class);
+    private UserService userService = UserServiceFactory.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        request.getRequestDispatcher("/users.jsp").forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        Long deleteUserId = Long.parseLong(request.getParameter("userId"));
+        Optional<User> deleteUser = userService.getUserById(deleteUserId);
```

Comment:
we can move this logic in the service layer

Reply:
we can write delete method which will find user by id and will remove it
so then in the servlet we don't need to call two methods: getUserById() & deleteUser()